src/syscalls.rs

```diff
@@ -35,6 +36,42 @@ pub fn yieldk() {
     }
 }
 
+#[cfg(target_arch = "riscv32")]
+pub fn yieldk() {
+    // Note: A process stops yielding when there is a callback ready to run,
+    // which the kernel executes by modifying the stack frame pushed by the
+    // hardware. The kernel copies the PC value from the stack frame to the LR
+    // field, and sets the PC value to callback to run. When this frame is
+    // unstacked during the interrupt return, the effectively clobbers the LR
+    // register.
+    //
+    // At this point, the callback function is now executing, which may itself
+    // clobber any of the other caller-saved registers. Thus we mark this
+    // inline assembly as conservatively clobbering all caller-saved registers,
+    // forcing yield to save any live registers.
+    //
+    // Upon direct observation of this function, the LR is the only register
+    // that is live across the SVC invocation, however, if the yield call is
+    // inlined, it is possible that the LR won't be live at all (commonly seen
+    // for the `loop { yieldk(); }` idiom) or that other registers are live,
+    // thus it is important to let the compiler do the work here.
+    //
+    // According to the AAPCS: A subroutine must preserve the contents of the
```

Comment:
This section (lines 59-62) appear to be ARM-specific. Are they leftovers?

Reply:
Yep, removed